src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -68,7 +68,21 @@
    :ruby-load-path           [schema/Str]
    :gem-path                 schema/Str
    :signeddir                schema/Str
-   :serial                   schema/Str})
+   :serial                   schema/Str
+   ;; This would be list of compile master node names (FQDNs) including MoM
+   ;; It would either be provisioned by PE or created & maintained by user (input configuration)
+   ;; OPEN: where do we get to specify defaults for the new configuration settings
+   ;; that will reside in ca.conf within certificate-authority?
+   ;; Would the installer be able to populate specifically the compile-masters with 
+   ;; current host name? 
+   :compile-masters          [schema/Str]    
+   ;; This would point to a file containing a list of serial numbers of compile master certificates
+   ;; corresponding to the nodes from the above file. This would be re-generated everytime the
+   ;; compile-masters list is updated.
+   ;; Q - is there a way to maintain this as an in-memory list?
```

Comment:
What do you mean?

Reply:
Currently during every revocation this serials file needs to be read. And while it should not have any perf implications I was thinking if there was a way to maintain the serials in memory at all times then that could be avoided.